### PR TITLE
Add SMN service

### LINKED ISSUE #20: Simple Message Notification
Implement `Simple Message Notification` service

- [x] Topic
  - [x] SDK
  - [x] CLI
  - [x] Docs
- [x] Subscription
  - [x] SDK
  - [x] CLI
  - [x] Docs
- [x] Template
  - [x] SDK
  - [x] CLI
  - [x] Docs
- [x] Message
  - [x] SDK
  - [x] CLI
  - [x] Docs
- [x] SMS
  - [x] SDK
  - [x] CLI
  - [x] Docs

### PR DESCRIPTION
**Pull request by tischrei**

### CODE REVIEW

File: doc/source/sdk/proxies/index.rst [hunk 1]
```diff
@@ -21,6 +21,7 @@ Service Proxies
    Object Block Storage (OBS) <obs>
    Relational Database Service RDS V1 (RDSv1) <rds_v1>
    Relational Database Service RDS V3 (RDS) <rds_v3>
+   Simple Message Notification Service (SMN) <smn>
```
Comment (lego963) [inline doc/source/sdk/proxies/index.rst:24]: is it ok that version is omitted?

File: otcextensions/osclient/smn/client.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+from osc_lib import utils
+
+from otcextensions import sdk
+from otcextensions.i18n import _
+
+LOG = logging.getLogger(__name__)
+
+DEFAULT_API_VERSION = '2'
+API_VERSION_OPTION = 'os_smn_api_version'
```
Comment (lego963) [inline otcextensions/osclient/smn/client.py:23]: maybe uppercase? if it's env var

File: otcextensions/osclient/smn/client.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+API_NAME = "smn"
+API_VERSIONS = {
+    "2": "openstack.connection.Connection"
```
Comment (lego963) [inline otcextensions/osclient/smn/client.py]: double quotes to single

**Review by lego963 — CHANGES_REQUESTED** [review]

File: doc/source/sdk/proxies/index.rst (hunk 1 above)
Comment (tischrei) [reply doc/source/sdk/proxies/index.rst:24]: yes, it is ok. Only in RDS it was necessary.

File: otcextensions/osclient/smn/client.py (hunk 2 above)
Comment (tischrei) [reply otcextensions/osclient/smn/client.py:23]: I keep the general format of the other services which was introduced by Artem. So I won't change it.

File: otcextensions/osclient/smn/client.py (hunk 3 above)
Comment (tischrei) [reply otcextensions/osclient/smn/client.py]: done

**Review by lego963 — DISMISSED** [review]

File: otcextensions/sdk/smn/v2/_proxy.py [hunk 4]
```diff
@@ -9,9 +9,306 @@
...
+        :param dict query: Optional query parameters to be sent to limit
+            the resources being returned.
+        :returns: A generator of subscription objects
+        :rtype: :class:`~otcextensions.sdk.smn.v2.subscription.Subscription`
+        """
+        if topic:
+            topic = self._get_resource(_topic.Topic, topic)
+            return self._list(_subscription.Subscription,
+                              topic_urn=topic.id, **query)
+        base_path = '/notifications/subscriptions'
```
Comment (anton-sidelnikov) [inline otcextensions/sdk/smn/v2/_proxy.py]: you can override list for subscriptions, for not to store this hardcoded parameter in proxy

**Review by anton-sidelnikov — DISMISSED** [review]

**Comment by anton-sidelnikov** [comment]
And also in one file you often use single quotes and double quotes for string parameters

**Review by irina-pereiaslavskaia — DISMISSED** [review]

**Review by Polina-Gubina — DISMISSED** [review]

File: otcextensions/sdk/smn/v2/_proxy.py (hunk 4 above)
Comment (tischrei) [reply otcextensions/sdk/smn/v2/_proxy.py]: But how can I solve the "if" cause then?

Comment (anton-sidelnikov) [reply otcextensions/sdk/smn/v2/_proxy.py]: in self._list check topic_urn, btw you can skip this comment, this is just idea)

**Comment by anton-sidelnikov** [comment]
@tischrei And where is func tests?

**Comment by tischrei** [comment]
> @tischrei And where is func tests?

I have never created func tests so far ^^

File: otcextensions/sdk/smn/v2/_proxy.py (hunk 4 above)
Comment (tischrei) [reply otcextensions/sdk/smn/v2/_proxy.py]: ok, done

**Review by lego963 — APPROVED** [review]

**Pull request merged**